### DISCUSSION TITLE
GitHub and GitLab integration is not working

### DISCUSSION THREAD

**Discussion by h4rithd** [GitbookIO/community#438] [3. Git Sync] [answered]
Hey team,

I am composing this message to address the issue with the integration of GitHub and GitLab. Despite attempting various methods multiple times, the problem persists. Each time I attempt to synchronize, I encounter the following error

<img width="999" alt="image" src="https://github.com/GitbookIO/community/assets/23568388/ce1c7f76-a2ef-436b-b556-423d14823b54">

**Comment by taranvohra** [comment]
Hey @h4rithd, can you post the ID of the space on which this is happening?

**Reply by h4rithd** [reply]
Hey it's `T******************WR`

**Reply by taranvohra** [reply, marked as answer]
I just had a look at the logs and turns out there was an issue at our side. It should work now. We're soon releasing a new git sync infrastructure that will be lot more resilient to these random errors.

**Reply by h4rithd** [reply]
Thank you for the support!

**Reply by DebbieAtSeam** [reply]
@taranvohra If the GitBook team has fixed this issue, but we are still seeing the GitHub sync error, what should we do to clear this error? Thanks!
![image](https://github.com/GitbookIO/community/assets/145377258/136c43f9-d65d-47c2-a46f-327935691084)

**Reply by taranvohra** [reply]
Hey @DebbieAtSeam, the issue should've resolved on it's own. If it's still there, can you please share the space `ID` so I can have a quick look at the logs?

**Reply by DebbieAtSeam** [reply]
Thanks, @taranvohra! You are correct. The issue did resolve itself. I confirmed our GitBook-GitHub sync in both directions. However, during the time period when I still saw the prominent failure message, it was a bit unnerving, especially because I wasn't quite sure how to fix the issue.

**Reply by taranvohra** [reply]
I totally understand @DebbieAtSeam! we're currently working on making some significant changes to the underlying Git Sync architecture for better reliability.

**Reply by kabirkbr** [reply]
i have experienced the problem explained above and it magically went away by itself; however, now I am observing unstable behaviour, with gitbook not reacting to a change in a gitlab repo, but only updating if manually triggered from 'GitLab Sync' / 'Settings' menu. Sometimes, however, the Gitlab sync plugin gets randomly deactivated and then activated without me doing anything.  At the current stage it is really difficult to use. 

waiting for those significant changes; when do you expect them to be released?
-- thanks!

**Comment by ahcsco** [comment]
This is a very buggy feature. It's still not working as expected. Keeps showing this error off and on. It's also having trouble finding repositories and branches in config settings. Also shows another strange error asking to install github on space even when its already installed and then complains that something went wrong because obviously it's already installed.
